src/index.ts

```diff
@@ -14,16 +14,19 @@
  * limitations under the License.
  */
 
-/* eslint require-await: 0 */
-'use strict'
+import { envSchema } from './env'
+import { getMetrics } from './metrics'
+import { UndecoratedService } from './types'
 
-import {DecoratedFastify} from "@mia-platform/custom-plugin-lib";
-const customService = require('@mia-platform/custom-plugin-lib')()
+const customService = require('@mia-platform/custom-plugin-lib')(envSchema)
 
-module.exports = customService(async function index(service: DecoratedFastify) {
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+module.exports = customService(async(service: UndecoratedService) => {
```

Comment:
shouldn't it be better to use the MainService rather then the Undecorated?

Reply:
Just a note: the use of `UndecoratedService` is due to this issue: https://github.com/mia-platform/custom-plugin-lib/issues/226 in the `custom-plugin-lib`.

Reply:
We use `UndecoratedService` since at plugin initialization no custom field has been decorated to the service yet. Using `MainService` would require either to set all custom decorated fields as optional or to have them already set (which is not possible).

Reply:
ok clear